sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/TransactionalBatchResponse.java

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.azure.cosmos.implementation.batch.BatchExecUtils;
+import com.azure.cosmos.util.Beta;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Response of a {@link TransactionalBatch} request.
+ */
+@Beta(Beta.SinceVersion.V4_7_0)
+public class TransactionalBatchResponse {
+
+    private final Map<String, String> responseHeaders;
+    private final int statusCode;
+    private final String errorMessage;
+    private final List<TransactionalBatchOperationResult<?>> results;
+    private final int subStatusCode;
+    private final CosmosDiagnostics cosmosDiagnostics;
+
+    /**
+     * Initializes a new instance of the {@link TransactionalBatchResponse} class.
+     *
+     * @param statusCode the response status code.
+     * @param subStatusCode the response sub-status code.
+     * @param errorMessage an error message or {@code null}.
+     * @param responseHeaders the response http headers
+     * @param cosmosDiagnostics the diagnostic
+     */
+    TransactionalBatchResponse(
+        final int statusCode,
+        final int subStatusCode,
+        final String errorMessage,
+        final Map<String, String> responseHeaders,
+        final CosmosDiagnostics cosmosDiagnostics) {
+
+        checkNotNull(statusCode, "expected non-null statusCode");
+        checkNotNull(responseHeaders, "expected non-null responseHeaders");
+
+        this.statusCode = statusCode;
+        this.subStatusCode = subStatusCode;
+        this.errorMessage = errorMessage;
+        this.responseHeaders = responseHeaders;
+        this.cosmosDiagnostics = cosmosDiagnostics;
+        this.results = new ArrayList<>();
+    }
+
+    /**
+     * Gets the result of the operation at the provided index in the current {@link TransactionalBatchResponse batch}.
+     * <p>
+     * @param <T> the type parameter.
+     * @param index 0-based index of the operation in the batch whose result needs to be returned.
+     * de-serialized, when present.
+     * @param type class type for which deserialization is needed.
+     *
+     * @return TransactionalBatchOperationResult containing the individual result of operation.
+     */
+    public <T> TransactionalBatchOperationResult<T> getOperationResultAtIndex(
```

Comment:
I think we should move the logic for decoding the ObjectNode from `TransactionalBatchResponse` to `TransactionalBatchOperationResult`.

I propose the following:
- we remove `TransactionalBatchResponse: getOperationResultAtIndex(int index, Class<T> type)` 
- we remove `TransactionalBatchOperationResult:getItem()`
- we add `TransactionalBatchOperationResult.getItem(Class<T> type)`

Reply:
@kirankumarkolli @FabianMeiswinkel @kushagraThapar please provide your input on this. Does this api look good to you?
I am still undecided on this one.

Reply:
The two options/patterns used are

- Indexed based correlation
- Iteration model with (request, response)
 
Between the two I am for #2, for Bulk #2 is used. For consistency also #2